poller/index.js

```diff
@@ -190,14 +190,28 @@ function postPubSubMessage(spanner, metrics) {
 async function parseAndEnrichPayload(payload) {
   var spanners = JSON.parse(payload);
 
-  for (var i = 0; i < spanners.length; i++) {
+  for (var sIdx = 0; sIdx < spanners.length; sIdx++) {
+    const metricOverrides = spanners[sIdx].metrics;
+
     // merge in the defaults
-    spanners[i] = {...spannerDefaults, ...spanners[i]};
-    spanners[i].metrics =
-        buildMetrics(spanners[i].projectId, spanners[i].instanceId);
-    spanners[i] = {
-      ...spanners[i],
-      ...await getSpannerMetadata(spanners[i].projectId, spanners[i].instanceId)
+    spanners[sIdx] = {...spannerDefaults, ...spanners[sIdx]};
+
+    spanners[sIdx].metrics =
+      buildMetrics(spanners[sIdx].projectId, spanners[sIdx].instanceId);
+    // merge in custom thresholds
+    if(metricOverrides != null) {
+      for (var oIdx = 0; oIdx < metricOverrides.length; oIdx++) {
+        mIdx = spanners[sIdx].metrics.findIndex(x => x.name === metricOverrides[oIdx].name);
```

Comment:
We could modify this loop to not only to override thresholds with custom ones but also to add custom metrics if the metric name is not found in the defaults.

Reply:
Yes, that is where we'll add custom metrics.  I would like to make other changes to make custom metrics implementation cleaner so in the spirit of keeping this PR small and focused just on thresholds, I was planning on doing that work in separate PR.